### PR TITLE
[XamlC] compile ListStringTypeConverter

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

compile the ListStringTypeConverter, used in Style and the new OnPlatform

### Behavioral Changes ###

slightly longer IL, but no reflection calls, and less method calls

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Build.Tasks/CompiledConverters/ListStringTypeConverter.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{
+	class ListStringTypeConverter : ICompiledTypeConverter
+	{
+		public IEnumerable<Instruction> ConvertFromString(string value, ModuleDefinition module, BaseNode node)
+		{
+			if (value == null) {
+				yield return Instruction.Create(OpCodes.Ldnull);
+				yield break;
+			}
+			var parts = value.Split(new[] { ',' }, StringSplitOptions.RemoveEmptyEntries).Select(s => s.Trim()).ToList();
```
Comment (samhouts) [inline Xamarin.Forms.Build.Tasks/CompiledConverters/ListStringTypeConverter.cs]: I wonder if there is a scenario where users would want a whitespace entry to occupy space in a list of strings. Perhaps providing labels for a chart where they want to skip every other item, for example?

Comment (samhouts) [inline Xamarin.Forms.Build.Tasks/CompiledConverters/ListStringTypeConverter.cs]: I wonder if there is a scenario where users would want a whitespace entry to occupy space in a list of strings. Perhaps providing labels for a chart where they want to skip every other item, for example?

File: Xamarin.Forms.Xaml.UnitTests/CompiledTypeConverter.xaml [hunk 2]
```diff
@@ -4,7 +4,8 @@
 			 x:Class="Xamarin.Forms.Xaml.UnitTests.CompiledTypeConverter"
 			 RectangleP="0,1,2,4"
 			 RectangleBP="4,8,16,32"
-			 BackgroundColor="Pink">
+			 BackgroundColor="Pink"
+			 List="Foo, Bar">
```
Comment (samhouts) [inline Xamarin.Forms.Xaml.UnitTests/CompiledTypeConverter.xaml]: If we're explicitly removing whitespace entries, should we test for that?

File: Xamarin.Forms.Build.Tasks/CompiledConverters/ListStringTypeConverter.cs (hunk 1 above)
Comment (samhouts) [inline Xamarin.Forms.Build.Tasks/CompiledConverters/ListStringTypeConverter.cs]: I wonder if there is a scenario where users would want a whitespace entry to occupy space in a list of strings. Perhaps providing labels for a chart where they want to skip every other item, for example?

Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/CompiledConverters/ListStringTypeConverter.cs]: All scenarios are possible... the ListStringTypeConverter (the new compiled version, and the old one) are only used for `VisualElement.StyleClass` where it is ok to RemoveEmptyEntries, but it's not required.

**Pull request merged**